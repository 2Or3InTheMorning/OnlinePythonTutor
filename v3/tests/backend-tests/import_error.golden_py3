{
  "code": "# should NOT allow for any imports\nimport os\n\nos.system(\"echo security breach\")\n", 
  "trace": [
    {
      "ordered_globals": [], 
      "stdout": "", 
      "func_name": "<module>", 
      "stack_to_render": [], 
      "globals": {}, 
      "heap": {}, 
      "line": 2, 
      "event": "step_line"
    }, 
    {
      "ordered_globals": [], 
      "stdout": "", 
      "exception_msg": "ImportError: os not supported\nOnly these modules can be imported into Python Tutor:\n  __future__, abc, array, bisect, calendar, cmath,\n  collections, copy, datetime, decimal, doctest, fractions,\n  functools, glob, hashlib, heapq, inspect, io,\n  itertools, json, locale, math, operator, pickle,\n  pprint, random, re, string, struct, time,\n  types, typing, unittest", 
      "func_name": "<module>", 
      "stack_to_render": [], 
      "globals": {}, 
      "heap": {}, 
      "line": 2, 
      "event": "exception"
    }
  ]
}
