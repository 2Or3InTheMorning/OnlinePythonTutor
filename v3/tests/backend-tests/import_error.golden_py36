{
  "code": "# should NOT allow for any imports\nimport os\n\nos.system(\"echo security breach\")\n",
  "trace": [
    {
      "line": 2,
      "event": "step_line",
      "func_name": "<module>",
      "globals": {},
      "ordered_globals": [],
      "stack_to_render": [],
      "heap": {},
      "stdout": ""
    },
    {
      "line": 2,
      "event": "exception",
      "func_name": "<module>",
      "globals": {},
      "ordered_globals": [],
      "stack_to_render": [],
      "heap": {},
      "stdout": "",
      "exception_msg": "ImportError: os not supported\nOnly these modules can be imported into Python Tutor:\n  __future__, abc, array, bisect, calendar, cmath,\n  collections, copy, datetime, decimal, doctest, fractions,\n  functools, glob, hashlib, heapq, inspect, io,\n  itertools, json, locale, math, operator, pickle,\n  pprint, random, re, string, struct, time,\n  types, typing, unittest"
    }
  ]
}
