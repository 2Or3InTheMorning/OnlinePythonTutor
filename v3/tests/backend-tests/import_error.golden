{
  "code": "# should NOT allow for any imports\nimport os\n\nos.system(\"echo security breach\")\n", 
  "trace": [
    {
      "ordered_globals": [], 
      "stdout": "", 
      "func_name": "<module>", 
      "stack_to_render": [], 
      "globals": {}, 
      "heap": {}, 
      "line": 2, 
      "event": "step_line"
    }, 
    {
      "ordered_globals": [], 
      "stdout": "", 
      "exception_msg": "ImportError: os not supported\nOnly these modules can be imported into Python Tutor:\n  StringIO, __future__, abc, array, bisect, calendar,\n  cmath, collections, copy, datetime, decimal, doctest,\n  fractions, functools, glob, hashlib, heapq, inspect,\n  io, itertools, json, locale, math, operator,\n  pickle, pprint, random, re, string, struct,\n  time, types, typing, unittest", 
      "func_name": "<module>", 
      "stack_to_render": [], 
      "globals": {}, 
      "heap": {}, 
      "line": 2, 
      "event": "exception"
    }
  ]
}
